### PR TITLE
[FLINK-1635] remove Apache Thrift and Google Protobuf dependencies

### PR DESCRIPTION
**Pull request by mxm**

### CODE REVIEW

**Comment by rmetzger** [comment]
It would be great if you could add some documentation that explains how users can register the serializers at the ExecutionConfig.

I've actually added support for these two frameworks because users needed this. It would be nice if your docs would explain how to use Flink with Thrift/Protobuf types (thats also needed for stuff like Parquet)

**Comment by StephanEwen** [comment]
Agree with Robert.

Still, this should go into the release, it gets several dependency issues out of the way.

**Comment by mxm** [comment]
Sure, adding documentation always makes sense :) I updated the pull request to include documentation on how to register custom serializers with Kryo.

File: docs/apis/best_practices.md [hunk 1]
```diff
@@ -155,3 +155,41 @@ public static final class Tokenizer extends RichFlatMapFunction<String, Tuple2<S
...
+
+final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+// register the Google Protobuf serializer with Kryo
+env.getConfig().registerTypeWithKryoSerializer(MyCustomType.class, ProtobufSerializer.class);
+
+// register the serializer included with Apache Thrift as the standard serializer
+// TBaseSerializer states it should be initalized as a default Kryo serializer
+env.getConfig().addDefaultKryoSerializer(MyCustomType.class, TBaseSerializer.class);
+env.getConfig().registerTypeWithKryoSerializer(MyCustomType.class, TMessage.class);
```
Comment (rmetzger) [inline docs/apis/best_practices.md]: this seems incorrect

Comment (mxm) [inline docs/apis/best_practices.md]: yes, sorry. copy/paste error. fixed.

**Comment by mxm** [comment]
I also added the necessary Maven configuration for the examples.

**Comment by rmetzger** [comment]
+1 to merge like this.

**Pull request closed without merging**